src/charm.py

```diff
@@ -824,6 +827,34 @@ def _push_tls_certificate_to_workload(self) -> None:
                 file_contents=internal_pem,
             )
 
+    def delete_tls_certificate_from_workload(self) -> None:
+        """Deletes TLS certificates from unit."""
+        external_ca, external_pem = self.tls.get_tls_files("unit")
+        if external_ca is not None:
+            delete_file_on_unit(
+                parent_dir=Config.MONGOD_CONF_DIR,
+                file_name=TLS_EXT_CA_FILE,
+            )
+
+        if external_pem is not None:
+            delete_file_on_unit(
+                parent_dir=Config.MONGOD_CONF_DIR,
+                file_name=TLS_EXT_PEM_FILE,
+            )
+
+        internal_ca, internal_pem = self.tls.get_tls_files("app")
+        if internal_ca is not None:
+            delete_file_on_unit(
+                parent_dir=Config.MONGOD_CONF_DIR,
+                file_name=TLS_INT_CA_FILE,
+            )
+
+        if internal_pem is not None:
+            delete_file_on_unit(
+                parent_dir=Config.MONGOD_CONF_DIR,
+                file_name=TLS_INT_PEM_FILE,
+            )
```

Comment:
Can't this be simplified to this? since you already test on the existence of the file on `delete_file_on_unit`

```suggestion
        delete_file_on_unit(parent_dir=Config.MONGOD_CONF_DIR, file_name=TLS_EXT_CA_FILE)
        delete_file_on_unit(parent_dir=Config.MONGOD_CONF_DIR, file_name=TLS_EXT_PEM_FILE)
        delete_file_on_unit(parent_dir=Config.MONGOD_CONF_DIR, file_name=TLS_INT_CA_FILE)
        delete_file_on_unit(parent_dir=Config.MONGOD_CONF_DIR, file_name=TLS_INT_PEM_FILE)
```

Reply:
fixed in https://github.com/canonical/mongodb-operator/pull/234/commits/9457f636d7faa7778bb4d7d9e1290e0bc34670fe